### ISSUE TITLE
Wrong perl used in test suite

### ISSUE THREAD

**Issue by eserte** [someone-stole-my-name/perl-nios#10]
On most of my smoker systems the test suite fails in different ways:
```
#   Failed test 'DNS/NIOS.pm loaded ok'
#   at t/00-compile.t line 46.
#          got: '65280'
#     expected: '0'
Major version specified as 2 - this is strictures version 1 at /usr/perl5.20.1Dp/lib/site_perl/5.20.1/strictures.pm line 17.
BEGIN failed--compilation aborted at /home/cpansand/.cpan/build/2021082822/DNS-NIOS-0.002-XtvA8A/blib/lib/DNS/NIOS.pm line 19.
Compilation failed in require at -e line 1.
...
```
or
```
Can't locate JSON.pm in @INC (you may need to install the JSON module) (@INC contains: /usr/home/cpansand/.cpan/build/2021082819/DNS-NIOS-0.002-1/blib/arch /usr/home/cpansand/.cpan/build/2021082819/DNS-NIOS-0.002-1/blib/lib /usr/local/lib/perl5/site_perl/mach/5.32 /usr/local/lib/perl5/site_perl /usr/local/lib/perl5/5.32/mach /usr/local/lib/perl5/5.32 .) at /usr/home/cpansand/.cpan/build/2021082819/DNS-NIOS-0.002-1/t/tlib/Test/NIOS.pl line 12.
BEGIN failed--compilation aborted at /usr/home/cpansand/.cpan/build/2021082819/DNS-NIOS-0.002-1/t/tlib/Test/NIOS.pl line 12.
Died at t/tlib/Test/SpawnNIOS.pm line 47.
...
```

It seems that the wrong perl executable is used here --- the first in user's PATH, which is usually /usr/bin/perl, and not the perl used for building the distribution. A possible fix is to call scripts in the test suite explicitly with `$^X`.

**Comment by someone-stole-my-name** [comment]
Thanks!, I hope both issues are fixed #11

**Comment by eserte** [comment]
0.004 looks fine.
